crates/uv-resolver/src/lock.rs

```diff
@@ -256,8 +256,31 @@ impl Distribution {
                     let built_dist = BuiltDist::Path(path_dist);
                     Dist::Built(built_dist)
                 }
-                // TODO: Handle other kinds of sources.
-                _ => todo!(),
+                SourceKind::Direct(direct) => {
+                    let filename: WheelFilename = self.wheels[best_wheel_index].filename.clone();
+
+                    // Reconstruct the URL with the subdirectory query parameter.
+                    let mut url = self.id.source.url.clone();
+                    if let Some(subdirectory) = direct.subdirectory.as_ref() {
+                        url.query_pairs_mut()
+                            .append_pair("subdirectory", subdirectory.as_str());
```

Comment:
Shouldn't this be fragment not query?

Reply:
Not necessarily. This format is internal to the lockfile. We’re using the fragment for the Git SHA.

Reply:
Isn't this method the reverse conversion, from lockfile to installable `Dist`?

Reply:
Yes you're right.